deployer/main.go

```diff
@@ -1,45 +1,94 @@
+        if reflect.ValueOf(clusterConfig.Server).IsZero() &&
+			reflect.ValueOf(clusterConfig.Token).IsZero() &&
+			reflect.ValueOf(clusterConfig.User).IsZero(){
+			//deployApp = false
+			infraContext := context.InitInfraContext()
+			beautifiedConfig, err := json.MarshalIndent(clusterConfig.Params, "", "   ")
+			if err != nil {
+				return
+			}
 
-	// create deployer context to hold global variables
-	deployerContext := context.InitDeployerContext(componentConfig.FlagArgs, kubeConnection)
+			log.Printf("About to deploy a cluster, clusterId:  %s , with following configuration (Every field that is not populated will be defaulted from source cluster): \n" , clusterConfig.ClusterId)
+			pretty.Printf("%s \n",string(beautifiedConfig))
+			newClusterDetails := provisioner.ProvisionCluster(infraContext, &clusterConfig.Params,clusterConfig.Aws, componentConfig.FlagArgs.RootSourceDir)
 
-	// check if application deployment has been requested
-	if !reflect.ValueOf(componentConfig.ApplicationConfig).IsZero() {
 
-		// create application deployment package
-		applicationPkg := packager.NewApplicationPkg(componentConfig.ApplicationConfig, deployerContext)
+			log.Printf("Successfully deployed a cluster, clusterId:  %s ",clusterConfig.ClusterId)
+			log.Printf("returned details of provisioned cluster with id : %s,  %+v\n" ,clusterConfig.ClusterId, newClusterDetails)
+			// If requested to deploy also the application on new cluster, then need to update kubeconfig with new details
 
-		// check if all mandatory variables have been set, else list unset vars and throw exception
-		if len(applicationPkg.UnsetMandatoryParams) > 0 {
-			log.Fatalf("ERROR: Please complete missing configuration for the following mandatory parameters (<FILEPATH>: <MANDATORY_PARAMETERS>.)\n%s", utils.StringifyMap(applicationPkg.UnsetMandatoryParams))
-		}
+			if !reflect.ValueOf(componentConfig.ApplicationConfig).IsZero(){
+				log.Print("Applying new Cluster address and Credentials , and kubeconfig file to be used by application deployment")
+				clusterProvisioned = true
+				clusterConfig.Server = newClusterDetails.ApiServer
+				clusterConfig.User = newClusterDetails.User
+				clusterConfig.Token = newClusterDetails.Password
+				kubeConfigPath = newClusterDetails.KubeConfigPath
+				authByToken = false
 
-		// deploy application deployment package
-		deployer.DeployApplication(applicationPkg)
+			}
 
-	} else {
-		log.Println("No application to deploy")
+		}
 	}
-}
+	 if deployApp {
+		pretty.Printf("Deploying the following configuration: \n%# v", componentConfig)
+		 // connect to cluster
+		 kubeConnection := connect.ConnectToCluster(clusterConfig, authByToken)
+		 // create deployer context to hold global variables
+		 deployerContext := context.InitDeployerContext(componentConfig.FlagArgs, kubeConnection)
+		 if clusterProvisioned {
+			 deployerContext.KubeConnection.KubeConfigPath = kubeConfigPath
```

Comment:
In line 65 (connect.ConnectToCluster) the kubeConfig is constructed and created as a temp file in the temp directory according to the available data.
Is it necessary to have this file created and set in the deployerContext just for it to be overwritten later in the line above?

Reply:
Didn't want to step in too much into your logic, so i'm overriding the kubeConfig File only after all your logic, only in case a cluster was provisioned, the password and user injection also should work, but by default for provisioned cluster doesn't support basic authentication, and the kube config generated by playbook contains already certificate, and ca locations, so this is necessary.

Reply:
Ok, That sounds like a good alternative to me